src/go/configgenerator/filterconfig/filter_generator.go

```diff
@@ -252,6 +252,15 @@ func updateProtoDescriptor(service *confpb.Service, apiNames []string, descripto
 					if method.GetOptions() == nil {
```

Comment:
`if rule, ok := ruleMap[sel]; ok {` is when http.rules in service config is defined but we want the case http.rules in proto is defined, which should check `method.GetOptions()`

When adding in additional_bindings, check if it is duplicate?

Reply:
Logic changed.